lib/Bindings/Python/circt/support.py

```diff
@@ -36,6 +36,38 @@ def get_value(obj) -> ir.Value:
   return None
 
 
+def var_to_attribute(obj) -> ir.Attribute:
+  """Create an MLIR attribute from a Python object for a few common cases."""
+  if isinstance(obj, ir.Attribute):
+    return obj
+  if isinstance(obj, int):
+    attrTy = ir.IntegerType.get_signless(64)
+    return ir.IntegerAttr.get(attrTy, obj)
+  if isinstance(obj, str):
+    return ir.StringAttr.get(obj)
+  if isinstance(obj, list):
+    return ir.ArrayAttr.get([var_to_attribute(x) for x in obj])
+  raise TypeError(f"Cannot convert type '{type(obj)}' to MLIR attribute")
+
+
+def attribute_to_var(attr: ir.Attribute):
```

Comment:
https://llvm.discourse.group/t/python-attribute-concreteattribute/3563

Reply:
I think the broader discussion is valuable and I'm +1 on helping push forward on that front, but for the task at hand, would `isinstance` work?

```
>>> a = mlir.ir.StringAttr.get("foo", context=ctx)
>>> isinstance(a, mlir.ir.Attribute)
True
>>> isinstance(a, mlir.ir.StringAttr)
True
>>> isinstance(a, mlir.ir.IntegerAttr)
False
```

I think you could use that to build a chain of `if isinstance(...)` with the appropriate constructor calls, rather than `try/except` around the direct constructor calls.